crates/red_knot/src/main.rs

```diff
@@ -120,9 +120,13 @@ fn run_check(args: CheckCommand) -> anyhow::Result<ExitStatus> {
     let system = OsSystem::new(cwd);
     let watch = args.watch;
     let exit_zero = args.exit_zero;
+    let cli_config_options = args.config.clone().into_options();
 
     let cli_options = args.into_options();
     let mut project_metadata = ProjectMetadata::discover(&project_path, &system)?;
+    if let Some(opts) = cli_config_options {
+        project_metadata.apply_cli_options(opts);
+    }
```

Comment:
We can only have one `cli_options` struct because it gets passed down to the `MainLoop` which applies the same options again when reloading the configuration in watch mode. 

I'd suggest to combine the `Options` inside `args.into_options` (which should be easily possible because `args` has access to `args.config`.

Reply:
Helpful context - thanks!